colcon-ros-cargo/colcon_ros_cargo/task/ament_cargo/build.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2018 Esteve Fernandez
+# Licensed under the Apache License, Version 2.0
+
+import os
+import shutil
+from pathlib import Path
+
+from colcon_cargo.task.cargo import CARGO_EXECUTABLE
+from colcon_cargo.task.cargo.build import CargoBuildTask
+
+from colcon_core.logging import colcon_logger
+from colcon_core.plugin_system import satisfies_version
+from colcon_core.shell import create_environment_hook
+from colcon_core.task import TaskExtensionPoint
+from colcon_core.task import run
+
+import toml
+
+
+logger = colcon_logger.getChild(__name__)
+
+# Some logic needs to be executed once per run.
+# There are no colcon hooks for this, so it is shoehorned into the build step
+# with a global.
+package_paths = None
+
+
+class AmentCargoBuildTask(CargoBuildTask):
+    """A build task for packages with Cargo.toml + package.xml.
+
+    The primary problem that needs to be solved is that dependencies on other
+    packages in the same workspace are expressed by just a name in colcon, but
+    by a full path in Cargo.
+
+    That means, when building a Cargo package, all the packages it depends on
+    need to be resolved to full paths, and those need to be written into the
+    Cargo.toml, or alternatively into a [patch] section of a .cargo/config.toml
+    file. Here the latter approach is used.
+    """
+
+    def __init__(self):  # noqa: D107
+        super().__init__()
+        satisfies_version(TaskExtensionPoint.EXTENSION_POINT_VERSION, '^1.0')
+
+    def add_arguments(self, *, parser):  # noqa: D102
+        parser.add_argument(
+            '--lookup-only-install',
```

Comment:
If I'm understanding this correctly. I think it should *only* search the install space, like what happens with every other package in a colcon workspace, that's why I asked in the chat if the source space was accessed in any way a few days ago. The source spaces of other packages are never accessed with the other build types, only the install spaces.

Reply:
Yep, I think this was a bit unfortunate: The two of us discussed this once in the past before some of the other people joined, and agreed that dependencies should point to the install space. Later, the topic was brought up again and people voted in favor of dependencies pointing to the source space by default, and I couldn't reach you for comment.

I changed the default now to searching only the install space. It's not a big deal since depending on the source space is still possible with the flag.

I also noticed that the README got lost at some point and added it back.

Reply:
No problem! That's what PR are for :slightly_smiling_face: